keystone_performance/load_test.py

```diff
@@ -304,6 +313,75 @@ def notify_done(self):
         self._done = True
 
 
+
+user_info_array = None
+
+def init_user_info(args):
+    global user_info_array
+    auth_url = '%s/v3/auth/tokens' % args.url
+
+    if args.user_domain_id:
+        user_domain_info = {'id': args.user_domain_id}
+    else:
+        user_domain_info = {'name': args.user_domain_name}
+    if args.project_id:
+        project_info = {'id': args.project_id}
+    else:
+        project_info = {'name': args.project_name}
+    if args.project_domain_id:
+        project_info['domain'] = {'id': args.project_domain_id}
+    else:
+        project_info['domain'] = {'name': args.project_domain_name}
+    auth_req_body = {
+        'auth': {
+            'identity': {
+                'methods': ['password'],
+                'password': {
+                    'user': {
+                        'name': args.username,
+                        'domain': user_domain_info,
+                        'password': args.password
+                    }
+                }
+            },
+            'scope': {
+                'project': project_info
+            }
+        }
+    }
+    auth_req_body = json.dumps(auth_req_body)
+
+    headers = {'content-type': 'application/json', 'Accept-Charset': 'UTF-8'}
+    r = requests.post(auth_url, data=auth_req_body, headers=headers)
+    if (r.status_code == 201):
+        token = r.headers['x-subject-token']
+    else:
+        raise Exception('Error: Failed to get keystone token for getting user info. Status code: ' + str(r.status_code))
+
+    users_url = '%s/v3/users?domain_id=default' % args.url
+    headers = {'content-type': 'application/json', 'X-Auth-Token': token}
+    r = requests.get(users_url, headers=headers)
+    if (r.status_code == 200):
+        users = json.loads(r.content)['users']
+    else:
+        raise Exception('Error: Failed to get user list. Status code: ' + str(r.status_code))
+
+    user_info = []
+    for user in users:
+        if user['name'].find(args.test_user_name_template) == 0:
+            user_info.append({'username': user['name'], 'domain_id': user['domain_id']})
+
+    user_info_array = user_info
+
+
+def get_random_user_info():
+    global user_info_array
+    if user_info_array == None:
+        raise Exception('Error: init_user_info must be called before get_random_user_info.')
```

Comment:
Change the code so that this can't happen, then this extra check doesn't need to be done.